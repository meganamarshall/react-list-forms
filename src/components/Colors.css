.Colors {
  div {
    height: 100px;
    width: 100px
  }
  .red {
    background-color: red;
  }
  .green {
    background-color: green;
  }
  .blue {
    background-color: blue;
    color: white;
  }
}
