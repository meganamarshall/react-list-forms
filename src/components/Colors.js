import React from 'react';
import PropTypes from 'prop-types';
import Color from './Color';

function Colors({ colors }) {
  const colorList = colors.map(color => {
    return (
      <li key={color.hex}>
        <Color name={color.name} hex={color.hex} />
      </li>
    );
  });

  return (
    <ul>
      {colorList}
    </ul>
  );
}

Colors.propTypes = {
  colors: PropTypes.array.isRequired
};

export default Colors;
