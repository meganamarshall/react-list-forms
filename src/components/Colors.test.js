import React from 'react';
import { shallow } from 'enzyme';
import Colors from './Colors';

describe('colors component', () => {
  it('renders the colors', () => {
    const colors = [{ name: 'red', rgb: { red: 255, green: 0, blue: 0 } }, { name: 'green', rgb: { red: 0, green: 255, blue: 0 } }, { name: 'blue', rgb: { red: 0, green: 0, blue: 255 } }];
    const wrapper = shallow(<Colors colors={colors} />);
    expect(wrapper).toMatchSnapshot();
  });
});
