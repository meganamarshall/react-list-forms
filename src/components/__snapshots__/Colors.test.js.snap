// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`colors component renders the colors 1`] = `
<ul>
  <li
    classList="name"
    key="undefined"
  >
    <Color
      name="red"
      rgb={
        Object {
          "blue": 0,
          "green": 0,
          "red": 255,
        }
      }
    />
  </li>
  <li
    classList="name"
    key="undefined"
  >
    <Color
      name="green"
      rgb={
        Object {
          "blue": 0,
          "green": 255,
          "red": 0,
        }
      }
    />
  </li>
  <li
    classList="name"
    key="undefined"
  >
    <Color
      name="blue"
      rgb={
        Object {
          "blue": 255,
          "green": 0,
          "red": 0,
        }
      }
    />
  </li>
</ul>
`;
